### PR TITLE
introduce timeouts config in types.Options

### LINKED ISSUE #5214: Group all timeouts in nuclei to single struct and derive from single value
### Please describe your feature request:

- In Nuclei to improve performance and handle unexpected situations , we have employed timeouts in multiple places for example
```console
MaxResponseHeaderTimeout -> [10 sec] -> time to wait for first byte of http response to be returned

ResponseReadTimeout (-rrt) -> [5 sec] -> network protocol conn read timeout

JsCompilerExecutionTimeout -> 1.5  x `-timeout` flag

HttpTimeout -> 3.0 x `-timeout` flag

DialTimeout -> `-timeout` flag

...
```
These heirarchical timeout handles conn hangs and other issues , while this is working and good , there is need to group all these together in a single struct say `TimeoutVariants` or something and allow configuring these from sdk

To improve UX, instead of adding flag for each timeout variant we should dervie all these values using multipliers from single flag i.e `-timeout`

### PR DESCRIPTION
**Pull request by dogancanbakir**
## Proposed changes

Closes #5214

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [ ] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: pkg/types/types.go [hunk 1]
```diff
@@ -401,6 +397,36 @@ type Options struct {
...
+		//timeout for network requests
+		DialTimeout: time.Second * time.Duration(options.Timeout),
+	}
+
+	if options.Timeout > 10 {
+		timeoutVariants.MaxResponseHeaderTimeout = time.Second * time.Duration(options.Timeout)
+	}
+
+	return timeoutVariants
+}
```
[the remark this answers is not in the document]
Comment (dogancanbakir) [reply pkg/types/types.go]: Might be missing something here, but the default is already 10 seconds. What do you mean by validating `options.Timeout`?

[the remark this answers is not in the document]
Comment (tarunKoyalwar) [reply pkg/types/types.go]: yes, it is 10seconds but when using sdk or internals ( which we do) there might be a possibility of Timeout being set to 0 . if i am not wrong we handle this explicitly in fastdialer or retryablehttp , thought we should do the same here

**Review by tarunKoyalwar — CHANGES_REQUESTED** [review]
changes + merge conflict + test failure

**Review by tarunKoyalwar — APPROVED** [review]
lgtm !

now all timeouts are stored in `Timeouts` struct which is pluggable ( not required and derived from -timeout but can manually be specified via SDK for more control ) .

All Known Timeouts and their default values
```go
// Timeouts is a struct that contains all the timeout variants for nuclei
// dialer timeout is used to derive other timeouts
type Timeouts struct {
	// DialTimeout for fastdialer (default 10s)
	DialTimeout time.Duration
	// Tcp(Network Protocol) Read From Connection Timeout (default 5s)
	TcpReadTimeout time.Duration
	// Http Response Header Timeout (default 10s)
	// this timeout prevents infinite hangs started by server if any
	// this is temporarily overridden when using @timeout request annotation
	HttpResponseHeaderTimeout time.Duration
	// HttpTimeout for http client (default -> 3 x dial-timeout = 30s)
	HttpTimeout time.Duration
	// JsCompilerExec timeout/deadline (default -> 2 x dial-timeout = 20s)
	JsCompilerExecutionTimeout time.Duration
	// CodeExecutionTimeout for code execution (default -> 3 x dial-timeout = 30s)
	CodeExecutionTimeout time.Duration
}
```

>[!CAUTION]
> This change removes below cli flags
> ```console
>  -dt, -dialer-timeout value  timeout for network requests.
>  -rrt, -response-read-timeout value  response read timeout in seconds (default 5s)
> ```

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**